batch_requests.go

```diff
@@ -0,0 +1,150 @@
+package msgraphgocore
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/google/uuid"
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+	"github.com/microsoftgraph/msgraph-sdk-go-core/internal"
+)
+
+// SendBatch sends a batch request
+func SendBatch(batch batchRequest, adapter abstractions.RequestAdapter) (BatchResponse, error) {
+	var res BatchResponse
+
+	batchJsonBody, err := batch.toJson()
+	if weGotAnError(err) {
+		return res, err
+	}
+
+	baseUrl, err := getBaseUrl(adapter)
+	if weGotAnError(err) {
+		return res, err
+	}
+
+	requestInfo := buildRequestInfo(batchJsonBody, baseUrl)
+	return sendBatchRequest(requestInfo, adapter)
+}
+
+// AppendBatchItem converts RequestInformation to a BatchItem and adds it to a BatchRequest
+//
+// You can add upto 20 BatchItems to a BatchRequest
+func (br *batchRequest) AppendBatchItem(reqInfo abstractions.RequestInformation) (*batchItem, error) {
+	if len(br.Requests) > 19 {
+		return nil, errors.New("Batch items limit exceeded. BatchRequest has a limit of 20 batch items")
+	}
+
+	batchItem, err := toBatchItem(reqInfo)
+	if weGotAnError(err) {
+		return nil, err
+	}
+
+	br.Requests = append(br.Requests, batchItem)
+	return batchItem, nil
+}
+
+// DependsOnItem creates a dependency chain between BatchItems.If A depends on B, then B will be sent before B
+func (bi *batchItem) DependsOnItem(item batchItem) {
+	// DependsOn is a single value slice
+	bi.DependsOn = []string{item.Id}
```

Comment:
can't you have a request depend on multiple other requests?

Reply:
No requests only have a single dependency.

Reply:
nothing would indicate that in the documentation, do you have sources to back this claim?
https://docs.microsoft.com/en-us/graph/json-batching#sequencing-requests-with-the-dependson-property

Reply:
https://docs.microsoft.com/en-us/graph/known-issues#request-dependencies-are-limited

Reply:
oh right, I missed that. thanks. But do we want to "propagate" the service limitation to the SDK and have to revisit the implementation should they fix this limitation?

Reply:
but nothing in the current implementation or doc comment indicates that a single dependency is supported. Right now this will erase any previously set dependency and if customers think they can set C depending on A and B, only B will be kept and C might execute before A.

Reply:
Adding this info to the doc comment.